bokeh/charts/_builder.py

```diff
@@ -190,49 +173,18 @@ def create(self, chart=None):
         if not chart.y_range:
             chart.y_range = self.y_range
 
-        # create the legends if needed
-        if self._legend:
-            if self._legend is True:
-                    orientation = "top_right"
-            else:
-                orientation = self._legend
-
-            legends = self._legends
-            chart.add_legend(orientation, legends)
+        # always contribute legends, let Chart sort it out
+        legends = self._legends
+        chart.add_legend(legends)
 
         return chart
 
     #***************************
     # Some helper methods
     #***************************
-    def _chunker(self, l, n):
-        """Yield successive n-sized chunks from l.
-
-        Args:
-            l (list: the incomming list to be chunked
-            n (int): lenght of you chucks
-        """
-        for i in range(0, len(l), n):
-            yield l[i:i + n]
-
-    def _set_colors(self, chunk):
-        """Build a color list just cycling through a defined palette.
-
-        Args:
-            chuck (list): the chunk of elements to generate the color list.
-        """
-        colors = []
-
-        pal = ["#f22c40", "#5ab738", "#407ee7", "#df5320", "#00ad9c", "#c33ff3"]
-        import itertools
-        g = itertools.cycle(pal)
-        for i in range(len(chunk)):
-            colors.append(next(g))
-
-        return colors
 
     def _set_and_get(self, data, prefix, attr, val, content):
-        """Set a new attr and then get it to fill the self.data dict.
+        """Set a new attr and then get it to fill the self._data dict.
```

Comment:
Private method, I'm inclined to just remove the docstring entirelty. I do want to clean up all this "set and get" business anyway, but not it this PR.